status/queues.py

```diff
@@ -755,10 +755,12 @@ def get(self):
             sample_dict['Reagent Label'] = geno_queries.get_reagentlabel_for_sample(geno_session, sampleid)
 
             #Get udfs specific to a step
-            if stepid in step_level_udfs_id:
-                step_level_udfs = geno_queries.get_sample_udfs_from_step(geno_session, sampleid, stepid, step_level_udfs_id[stepid])
-                for udfname, udfvalue, _ in step_level_udfs:
-                    sample_dict[udfname] = udfvalue 
+            for step in stepid_to_stepindex:
+                if stepid_to_stepindex[step]<= stepid_to_stepindex[stepid]:
+                    if step in step_level_udfs_id:
+                        step_level_udfs = geno_queries.get_sample_udfs_from_step(geno_session, sampleid, step, step_level_udfs_id[step])
+                        for udfname, udfvalue, _ in step_level_udfs:
+                            sample_dict[udfname] = udfvalue
```

Comment:
Do you know it's not already set, or do you not care?

Reply:
How do you mean? Set for a udf of the same name in a different step?

Reply:
Yes exactly

Reply:
The udfs are mostly unique to the steps we check for. Otherwise we could have a mapping of the relevant udf-step combinations we want to extract.

Reply:
So we don't really care about overwriting them? I guess it could be OK

Reply:
I put in a warning to the log if there are two values for a single udf in a step